airbyte-integrations/connectors/source-shopify/source_shopify/source.py

```diff
@@ -292,20 +338,76 @@ def request_params(
         return params
 
 
+class MetafieldOrders(MetafieldShopifySubstream):
+    parent_stream_class: object = Orders
+
+
 class DraftOrders(IncrementalShopifyStream):
     data_field = "draft_orders"
 
     def path(self, **kwargs) -> str:
         return f"{self.data_field}.json"
 
 
+class MetafieldDraftOrders(MetafieldShopifySubstream):
+    parent_stream_class: object = DraftOrders
+
+
 class Products(IncrementalShopifyStream):
     data_field = "products"
 
     def path(self, **kwargs) -> str:
         return f"{self.data_field}.json"
 
 
+class MetafieldProducts(MetafieldShopifySubstream):
+    parent_stream_class: object = Products
+
+
+class ProductImages(ShopifySubstream):
+    parent_stream_class: object = Products
+    slice_key = "id"
+    data_field = "images"
+
+    def path(self, stream_slice: Mapping[str, Any] = None, **kwargs) -> str:
+        product_id = stream_slice[self.slice_key]
+        return f"products/{product_id}/{self.data_field}.json"
+
+
+class MetafieldProductImages(MetafieldShopifySubstream):
+    parent_stream_class: object = Products
+    nested_record = "images"
+    slice_key = "images"
+    nested_substream = "images"
+    nested_substream_list_field_id = "id"
+
+    def path(self, stream_slice: Mapping[str, Any] = None, **kwargs) -> str:
+        image_id = stream_slice[self.slice_key]
+        return f"product_images/{image_id}/{self.data_field}.json"
+
+
+class ProductVariants(ShopifySubstream):
```

Comment:
Same situation here with `variants`, it's the part of `product` record, we can reuse `nested_substream = "variants"` to save the sync time. Correct me if I'm wrong.

Reply:
agreed